### PR TITLE
Add clang-format to CI, PR template, etc

### PR DESCRIPTION
**Pull request by matthew-reynolds**
# Pull Request

This is a step towards enforcing code style. The PR template will now ask that you have auto formatted your code, and CI will fail if code is non-conforming. Also added a bit more info about code style to the README. Currently, only the `clang-format` is checked in CI, since we don't have rules for `yapf` or `clang-tidy`.

## Contribution Checklist

- [x] I have read the [contributing](https://github.com/uwreact/frc_control/blob/melodic-devel/CONTRIBUTING.md) guide.
- [x] I have referenced relevant issues.
- [x] I have labeled accordingly.
- [x] I have detailed my changes as much as possible.

## Change Checklist

- [x] I have added tests.
- [x] I have added necessary documentation.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -7,8 +7,20 @@ IndentWidth:        2
 UseTab:             Never
 BreakBeforeBraces:  Custom
 BraceWrapping:
-  BeforeCatch: true
-  BeforeElse: true
+  AfterClass:             false
```
Comment (wmmc88) [inline .clang-format:10]: I think this should be true

File: README.md [hunk 2]
```diff
@@ -18,6 +18,19 @@ Once the workspace has been setup for cross-compilation, you can native or cross
...
+
+We largely follow the ROS coding guidelines, with a few noteable exceptions. To make development easy, we provide configuration files for standard linting and static analysis tools such as [clang-format](https://clang.llvm.org/docs/ClangFormat.html), [clang-tidy](https://clang.llvm.org/extra/clang-tidy) and [yapf](https://github.com/google/yapf).
+
+Install the linters:
+
+    $ sudo apt install clang-format-7 clang-tidy-7 yapf
+
+To run the linters:
+
+    $ find . -name "*.h" -o -name "*.cpp" | xargs clang-format-7 -i -style=file
```
Comment (wmmc88) [inline README.md:31]: Maybe clarify clang is for c and yapf for python

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -23,3 +23,4 @@ While we look over every pull request, we maintain a focus on this project's cur
 
 - [ ] I have added tests.
 - [ ] I have added necessary documentation.
+- [ ] I have auto formatted using clang-format or yapf.
```
Comment (wmmc88) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: clang-format/yapf might sound better lol idk

**Review by wmmc88 — APPROVED** [review]
Small nitpicky things I commented on but good to merge

File: .clang-format (hunk 1 above)
Comment (matthew-reynolds) [reply .clang-format:10]: Nah the example documentation is just wrong. This is the behaviour:

True:
```
class MyClass
{
  // blah
}
```

False:
```
class MyClass {
  // blah
}
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (matthew-reynolds) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: Gonna leave this for now and we can iron out the exact language when we also add the clang-tidy

File: README.md (hunk 2 above)
Comment (matthew-reynolds) [reply README.md:31]: Don't think that that's really neccessary? But again, we will clarify this exact language when we also add the clang-tidy

**Pull request merged**